addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiSensorWaterHandler.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mihome.handler;
+
+import static org.openhab.binding.mihome.XiaomiGatewayBindingConstants.CHANNEL_LEAK;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Thing;
+
+import com.google.gson.JsonObject;
+
+/**
+ * Handles the Xiaomi water leak sensor
+ *
+ * @author Kuba Wolanin - Initial contribution
+ */
+public class XiaomiSensorWaterHandler extends XiaomiSensorBaseHandler {
+
+    @Override
+    void parseReport(JsonObject data) {
+        boolean leak = data.has("status") && "leak".equals(data.get("status").getAsString());
+        boolean noLeak = data.has("status") && "no_leak".equals(data.get("status").getAsString());
```

Comment:
is there a state, where there is no "no_leak" and no "leak"?
this solution seems a bit odd. why not just have boolean leak = false --> no leak
from my point of view there is only the need to check for "leak", else set false.